src/sentry/projects/services/project/impl.py

```diff
@@ -35,6 +35,19 @@ def get_by_id(self, *, organization_id: int, id: int) -> RpcProject | None:
             return serialize_project(project)
         return None
 
+    def get_by_slug(self, *, organization_id: int, slug: str) -> RpcProject | None:
+        try:
+            project: Project | None = Project.objects.get_from_cache(
+                slug=slug, organization=organization_id
+            )
+        except ValueError:
+            project = Project.objects.filter(slug=slug, organization=organization_id).first()
```

Comment:
I copied the cache lookup from `get_by_id`. @vgrozdanic do you know more about it? Is is relevant here?

Reply:
I think this is not needed here, since it will always raise `ValueError`, because project slug is not one of the cache fields: https://github.com/getsentry/sentry/blob/c2464769739ea2f2bb84ba5380b471313d99620f/src/sentry/models/project.py#L343

Cache fields are defined when declaring the ORM manager for the model (it is responsible for talking with the DB), and `get_from_cache` [will raise an exception](https://github.com/getsentry/sentry/blob/c2464769739ea2f2bb84ba5380b471313d99620f/src/sentry/db/models/manager/base.py#L297) if you are trying to query by using one of the fields not defined in the `cache_fields`

Reply:
thx for the explanation! 👍

Reply:
You can just use:
```
 project: Project | None = Project.objects.filter(slug=slug, organization=organization_id).first()
```
since `.first()` method won't raise an exception (there is no need to catch ` Project.DoesNotExist`), it will either return an object or `None` if nothing is found